### PR TITLE
Ftl readme

### PR DESCRIPTION
**Pull request by tikurahul**
Adds a detailed FTL guide to the Macrobenchmark samples.

### CODE REVIEW

File: MacrobenchmarkSample/ftl/README.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
+# Setting up Jetpack Macrobenchmarks for CI
+
+[Jetpack Macrobenchmark](https://d.android.com/benchmark) is a Jetpack Library with Android Studio integration that allows app developers to measure performance of startup, scrolling, and other user interactions on real devices. The goal is to both enable more accurate and consistent measurements for local profiling, as well as in CI.
+
+One of the goals of the project is to provide samples and documentation to enable app developers to easily integrate Macrobenchmarks into their CI.
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:5]: we probably dont need to explain a goal of the project

File: MacrobenchmarkSample/ftl/README.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  - Set up custom dashboards
+  - Set up alerts
+
+## Project Setup
+
+```
+You will need a Google Account to setup your Google Cloud Platform project.
+```
+
+If you don't have a Firebase project for your app, go to the [Firebase Console](https://console.firebase.google.com/u/0/?pli=1) and click `Create New Project` to create one now. You will need ownership or edit permissions in your project. You might also want to upgrade to the [Blaze plan](https://firebase.google.com/docs/test-lab/usage-quotas-pricing), if you run out of [free quota](https://firebase.google.com/docs/test-lab/usage-quotas-pricing).
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:28]: `to create a new one.`

File: MacrobenchmarkSample/ftl/README.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+Now that you have a service account, you also need to grant the service account access to the Firebase Test Lab APIs. For more information, please refer to [enabling an API for your project](https://cloud.google.com/endpoints/docs/openapi/enable-api).
+
+#### Create a Cloud Storage Bucket
+
+[Create](https://cloud.google.com/storage/docs/creating-buckets) a Google Cloud Storage bucket in your project called `macrobenchmark-results`. This will be the location where the results of test runs from Firebase Test Lab are stored. 
+
+#### Deploy the Google Cloud Function
+
+A [Google Cloud function](#TODO) is used to process the results of test runs from Firebase Test Lab, and process the Macrobenchmark results `json` file. The cloud function also captures all the quantitative metrics from the `json` file, and uploads them to Google Cloud Monitoring.
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:51]: TODO?

File: MacrobenchmarkSample/ftl/README.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+The cloud function uses the [default service account](https://cloud.google.com/functions/docs/securing/function-identity) associated with the Google Cloud Platform project. This service account should be assigned roles that give it access to Google Cloud Storage (so it has access to the created buckets and results stored by Firebase Test Lab), and Google Cloud Monitoring. For more information, please refer to [enabling an API for your project](https://cloud.google.com/endpoints/docs/openapi/enable-api).
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:53]: unrelated to this PR but technically, the github action could do this as well, no? (pull the result and expert data into monitoring?)

File: MacrobenchmarkSample/ftl/README.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+### GitHub Workflow
+
+You can base your GitHub workflow, based on [this](https://github.com/android/performance-samples/blob/macrobenchmark/.github/workflows/macrobenchmark.yml) sample.
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:57]: You can `create` your Github Workflow...

File: MacrobenchmarkSample/ftl/README.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+
+```yaml
+name: Build and Run Macrobenchmarks
+on:
+  push:
+    branches: [ macrobenchmark ]
+  workflow_dispatch:
+```
+
+This part of the workflow defines the events that trigger a Macrobenchmark run. For our workflow we want to trigger a workflow when a **pull request is merged** or **pushed** to a branch `macrobenchmark`.
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:124]: to the branch `macrobenchmark`

File: MacrobenchmarkSample/ftl/README.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+
+#### One Time Setup
+
+##### Deploying the cloud function
+
+The first step is to deploy the Google Cloud Functions to your GCP project. Make sure you have downloaded the `gcloud` [CLI])(https://cloud.google.com/sdk/gcloud) and the Firebase [CLI](https://firebase.google.com/docs/cli) setup. 
+
+Copy the cloud function from [here](https://github.com/android/performance-samples/tree/macrobenchmark/MacrobenchmarkSample/functions).
+
+Then, initialize your project using the below commands.
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:205]: using the commands below.

File: MacrobenchmarkSample/ftl/README.md [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+All Macrobenchmark results are stored as [custom metrics](https://cloud.google.com/monitoring/custom-metrics) using Google Cloud Monitoring APIs. These metrics have a namespace `custom.googleapis.com` so you can use the [Metrics Query Editor](https://cloud.google.com/monitoring/mql/query-editor) to query and create dashboards and alerts to catch regressions.
+
+For more information on how you can use the MQL (query language) to create alerts, please refer to the [following documentation](https://cloud.google.com/monitoring/mql/alerts).
+
+#### Monitoring Dashboard
+
+![Monitoring Dashboard](./images/monitoring.png)
+
+#### Example Metrics
+
```
Comment (yigit) [inline MacrobenchmarkSample/ftl/README.md:289]: was there supposed to be another image here?

File: MacrobenchmarkSample/ftl/README.md (hunk 4 above)
Comment (tikurahul) [reply MacrobenchmarkSample/ftl/README.md:53]: That is correct. We originally wanted FTL team to own this and provide this as a service. But they declined. We could simplify this flow in the future, and make this an action.

File: MacrobenchmarkSample/ftl/README.md (hunk 8 above)
Comment (tikurahul) [reply MacrobenchmarkSample/ftl/README.md:289]: I added 2 images originally. The second one was kind of redundant so I got rid of it. 
Removed a title block to simplify this part.

**Pull request merged**